generator/utils.go

```diff
@@ -0,0 +1,282 @@
+package generator
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/onflow/cadence/runtime/ast"
+	"github.com/onflow/cadence/runtime/cmd"
+	"github.com/onflow/cadence/runtime/common"
+	"github.com/onflow/flixkit-go"
+	"github.com/onflow/flow-cli/flowkit"
+	"github.com/onflow/flow-go-sdk"
+)
+
+func processParameters(program *ast.Program, code string, template *flixkit.FlowInteractionTemplate) error {
+	template.Data.Arguments = make(flixkit.Arguments)
+	if program.SoleTransactionDeclaration() != nil {
+		if program.SoleTransactionDeclaration().ParameterList != nil {
+			for i, param := range program.SoleTransactionDeclaration().ParameterList.Parameters {
+				template.Data.Arguments[param.Identifier.String()] = flixkit.Argument{
+					Type:  param.TypeAnnotation.Type.String(),
+					Index: i,
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+// Support FLIP - Interaction Template Cadence Doc
+// # Interaction Template Cadence Doc (v1.0.0)
+// https://github.com/onflow/flips/pull/80
+func processCadenceCommentBlock(cadenceCode string, template *flixkit.FlowInteractionTemplate) error {
+	commentBlockPattern := regexp.MustCompile(`/\*[\s\S]*?@f_version[\s\S]*?\*/`)
+	codeCommentBlock := commentBlockPattern.FindString(cadenceCode)
+	template.Data.Cadence = cadenceCode
+	fType, err := determineCadenceType(cadenceCode)
+
+	if err != nil {
+		return err
+	}
+	template.Data.Type = fType
+	if fType == "interface" {
+		template.FType = "InteractionTemplateInterface"
+	}
+
+	// no comment block found
+	if codeCommentBlock == "" {
+		return nil
+	}
+
+	versionRE := regexp.MustCompile(`\s*@f_version (.+)`)
```

Comment:
On a general basis would like to state my objection to use regex for parsing.  

Can we look at prior work here for cadence-doc parsing? https://github.com/onflow/cadence-tools/tree/master/docgen

Reply:
I'd second what @bjartek said. I tried to do something I thought was simple before in Cadence and @jribbink pointed out tens of issues I hadn't realized would be issues without proper parsing. Decided to close the PR and wait for a parsing option because it introduced more problems than it solved. The correct way to to do it is probably to have the Cadence lang support (via the parser) whatever was introduced in a FLIP. You also don't want them being out of sync with what is happening.

Reply:
Also added a way for user to pass in a pre populated template in case they don't want to use cadence comment block.

Reply:
I'll strip out Cadence Comment block stuff, user can pass in pre-filled template.